test/UnitTests/AccessTokenHandlerTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using IdentityModel.Client;
+using Xunit;
+
+namespace IdentityModel.UnitTests
+{
+    public class AccessTokenHandlerTests
+    {
+        [Fact]
+        public async Task The_401_response_that_causes_token_refresh_and_retry_should_be_disposed_to_unblock_socket()
+        {
+            var document = File.ReadAllText(FileName.Create("success_token_response.json"));
+            var handler = new NetworkHandler(document, HttpStatusCode.OK);
+
+            using (var tokenClient = new TokenClient(
+                "http://server/token",
+                "client",
+                handler))
+            {
+                var indirectOutputOfHttpResponses = new StubHttpResponsesHandler();
+                var accessTokenHandler = new AccessTokenHandler(
```

Comment:
Slight difference in this test compared to the `RefreshTokenHandlerTests` - I don't first get the `AccessToken`. This actually has higher code coverage and I expect users to not bother with getting the access token themselves anyway.